src/mame/drivers/gi6809.cpp

```diff
@@ -0,0 +1,1039 @@
+	PORT_BIT( 0x20, IP_ACTIVE_LOW, IPT_UNUSED )
+	PORT_BIT( 0x40, IP_ACTIVE_LOW, IPT_UNUSED )
+	PORT_BIT( 0x80, IP_ACTIVE_LOW, IPT_UNUSED )
+
+	PORT_START("IN0-3")
+	PORT_BIT( 0x01, IP_ACTIVE_LOW, IPT_POKER_HOLD1 )  PORT_NAME("1 Coin Start (1 Line) / Up (Test Menu)")
+	PORT_BIT( 0x02, IP_ACTIVE_LOW, IPT_POKER_HOLD2 )  PORT_NAME("2 Coin Start (2 Lines) / Down (Test Menu)")
+	PORT_BIT( 0x04, IP_ACTIVE_LOW, IPT_POKER_HOLD5 )  PORT_NAME("5 Coin Start (5 Lines)")
+	PORT_BIT( 0x08, IP_ACTIVE_LOW, IPT_POKER_HOLD4 )  PORT_NAME("4 Coin Start (4 Lines)")
+	PORT_BIT( 0x10, IP_ACTIVE_LOW, IPT_POKER_HOLD3 )  PORT_NAME("3 Coin Start (3 Lines)")
+	PORT_BIT( 0x20, IP_ACTIVE_LOW, IPT_UNUSED )
+	PORT_BIT( 0x40, IP_ACTIVE_LOW, IPT_UNUSED )
+	PORT_BIT( 0x80, IP_ACTIVE_LOW, IPT_UNUSED )
+
+	PORT_START("PIA0_A")
+	PORT_BIT( 0x01, IP_ACTIVE_HIGH, IPT_OTHER )  PORT_NAME("PIA0_A 01")   // Meter Sensor 1.
+	PORT_BIT( 0x02, IP_ACTIVE_HIGH, IPT_OTHER )  PORT_NAME("PIA0_A 02")   // Meter Sensor 2.
+	PORT_BIT( 0x04, IP_ACTIVE_HIGH, IPT_OTHER )  PORT_NAME("PIA0_A 04")   // Meter Sensor 3.
+	PORT_BIT( 0x08, IP_ACTIVE_HIGH, IPT_OTHER )  PORT_NAME("PIA0_A 08")   // Meter Sensor 4.
+	PORT_BIT( 0x10, IP_ACTIVE_HIGH, IPT_OTHER )  PORT_NAME("PIA0_A 10")   // Meter Sensor 5.
+	PORT_BIT( 0x20, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x40, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x80, IP_ACTIVE_LOW, IPT_UNKNOWN )
+
+	PORT_START("SENSB")
+	PORT_BIT( 0x01, IP_ACTIVE_LOW, IPT_OTHER )   PORT_NAME("Coin Mech")       PORT_CODE(KEYCODE_H)
+	PORT_BIT( 0x02, IP_ACTIVE_LOW, IPT_OTHER )   PORT_NAME("Hopper Sensor")   PORT_CODE(KEYCODE_J)
+	PORT_BIT( 0x04, IP_ACTIVE_LOW, IPT_OTHER )   PORT_NAME("Cashbox Switch")  PORT_CODE(KEYCODE_K)
+	PORT_BIT( 0x08, IP_ACTIVE_LOW, IPT_OTHER )
+	PORT_BIT( 0x10, IP_ACTIVE_LOW, IPT_OTHER )
+	PORT_BIT( 0x20, IP_ACTIVE_LOW, IPT_OTHER )
+	PORT_BIT( 0x40, IP_ACTIVE_LOW, IPT_OTHER )
+	PORT_BIT( 0x80, IP_ACTIVE_LOW, IPT_OTHER )
+
+INPUT_PORTS_END
+
+
+/*********************************************
+*              Graphics Layouts              *
+*********************************************/
+
+static const gfx_layout tilelayout =
+{
+	8, 8,
+	RGN_FRAC(1,3),
+	3,
+	{ 0, RGN_FRAC(1,3), RGN_FRAC(2,3) },
+	{ 0, 1, 2, 3, 4, 5, 6, 7 },
+	{ 0*8, 1*8, 2*8, 3*8, 4*8, 5*8, 6*8, 7*8 },
+	8*8
+};
+
+
+/**************************************************
+*           Graphics Decode Information           *
+**************************************************/
+
+static GFXDECODE_START( gfx_gi6809 )
+	GFXDECODE_ENTRY( "gfx1", 0, tilelayout, 0, 16 )
+	GFXDECODE_ENTRY( "gfx2", 0, tilelayout, 0, 16 )
+GFXDECODE_END
+
+
+/**************************************************
+*            Discrete Sound Circuitry             *
+**************************************************/
+
+static const discrete_555_desc gi6809_555_vco_desc =
+	{DISC_555_OUT_DC | DISC_555_OUT_ENERGY, 5, DEFAULT_555_VALUES};
+
+static const discrete_dac_r1_ladder dac_gi6809_ladder =
+{
```

Comment:
It’s time we stop accepting new code using the legacy discrete sound.  Please migrate this to the netlist system.